torch/csrc/jit/script/compiler.cpp

```diff
@@ -784,9 +809,29 @@ Value* emitBuiltinCall(
   if (!required) {
     return nullptr;
   }
-  if(variants.size() == 0) {
-    throw ErrorReport(loc) << "unknown builtin op";
+
+  // no operators found with the same name, print out similarly named operators
+  if (variants.size() == 0) {
+    auto close_operators = findOperatorsWithSimilarName(name);
+    auto error = ErrorReport(loc);
+
+    std::string user_function_name(name.toQualString());
+    error << "unknown builtin op: " << user_function_name << "\n";
+    if (close_operators.size() == 0) {
+      error << "Could not find any similar ops to " << user_function_name << ". This op may not exist or may not be currently supported in TorchScript\n";
+    } else {
+      error << "Here are some suggestions: \n";
+      auto best_dist = (*close_operators.begin()).first;
+      for (auto rank: close_operators) {
+        if (rank.first > best_dist + 1) {
```

Comment:
I don't think you need select and rank the operators in two different functions. Why can't there be a single function that returns the `n` closest ops? Doing that with a priority queue with the comparator being edit distance is a more explicit way to accomplish what you're doing here.